src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypingEngine.java

```diff
@@ -339,27 +325,18 @@ private OutputAlleleSubset(final int count, final Allele[] alleles, final int[]
         }
 
         private List<Allele> outputAlleles(final Allele referenceAllele) {
-            final ArrayList<Allele> result = new ArrayList<>(count + 1);
-            result.add(referenceAllele);
-            for (int i = 0; i < count; i++) {
-                result.add(alleles[i]);
-            }
-            return result;
+            return Stream.concat(Stream.of(referenceAllele), Arrays.stream(alleles, 0, count)).collect(Collectors.toList());
```

Comment:
I'll just take this opportunity to say that we have no policy in GATK4 mandating that all loops be replaced with Java 8-style one-liners, and that, in my experience, sometimes the streaming version is clearer/more readable, and other times the old-school loop is clearer/more readable. There's no shame in a for loop -- we should always prefer the clearer/more straightforward version, whatever that may be.

Now, if you think the streaming version is more readable here, then I defer to your judgment despite personally thinking that the old-fashioned loop has the slight readability edge here :)